File: test/python/dsa/myarrayqueuetest.py
import sys
import unittest

sys.path.append("src/python")
from dsa.myarrayqueue import MyArrayQueue


class MyArrayQueueTest(unittest.TestCase):
    @classmethod
    def setUpClass(cls) -> None:
        return super().setUpClass()

    def setUp(self) -> None:
        self.myArrayQueue = MyArrayQueue[int]()
        self.myArrayQueue.clear()

    def test_equals(self):
        self.assertFalse(self.myArrayQueue.equals(self.myArrayQueue))

    def test_isEmpty(self):
        self.assertTrue(self.myArrayQueue.isEmpty())

    def test_offer(self):
        self.myArrayQueue.offer(5)
        self.assertIsNone(self.myArrayQueue.peek())

    def test_peek(self):
        self.assertIsNone(self.myArrayQueue.peek())

    def test_poll(self):
        self.assertIsNone(self.myArrayQueue.poll())

    def test_size(self):
        self.assertEqual(0, self.myArrayQueue.size())

    def test_toString(self):
        self.assertEqual("[]", self.myArrayQueue.toString())

    def tearDown(self) -> None:
        return super().tearDown()

    @classmethod
    def tearDownClass(cls) -> None:
        return super().tearDownClass()


if __name__ == "__main__":
    unittest.main()

File: test/typescript/dsa/myarrayqueue.test.ts
import MyArrayQueue from "../../../src/typescript/dsa/myarrayqueue";

let myArrayQueue: MyArrayQueue<number>;

beforeAll(() => {

});

beforeEach(() => {
    myArrayQueue = new MyArrayQueue<number>();
    myArrayQueue.clear();
});

test("Test constructor", () => {
    expect(() => {
        new MyArrayQueue<number>(-1);
    }).toThrow(Error);
});

test("Test equals", () => {
    expect(myArrayQueue.equals(myArrayQueue)).toBeFalsy();
});

test("Test isEmpty", () => {
    expect(myArrayQueue.isEmpty()).toBeTruthy();
});

test("Test offer", () => {
    myArrayQueue.offer(5);
    expect(myArrayQueue.peek()).toBeNull();
});

test("Test peek", () => {
    expect(myArrayQueue.peek()).toBeNull();
});

test("Test poll", () => {
    expect(myArrayQueue.poll()).toBeNull();
});

test("Test size", () => {
    expect(myArrayQueue.size()).toBe(0);
});

test("Test toString", () => {
    expect(myArrayQueue.toString()).toBe("[]");
});

afterEach(() => {

});

afterAll(() => {

});
